core/optaplanner-core-impl/src/main/java/org/optaplanner/core/api/domain/variable/NextElementShadowVariable.java

```diff
@@ -0,0 +1,34 @@
+package org.optaplanner.core.api.domain.variable;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.optaplanner.core.api.domain.entity.PlanningEntity;
+import org.optaplanner.core.api.solver.Solver;
+
+/**
+ * Specifies that a bean property (or a field) is an element in the same {@link PlanningListVariable} with an index that is
+ * higher by 1 than this element's index. May be {@code null} if this is the last element in the list variable.
```

Comment:
```suggestion
 * Specifies that a bean property (or a field) references the next element in the same {@link PlanningListVariable}.
 * So that element's index is higher by 1 than this element's index.
 * It is {@code null} if this element is the last element in the list variable.
```